voice-search/voice-search-impl/src/main/java/com/duckduckgo/voice/impl/VoiceSearchActivityLauncher.kt

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.voice.impl
+
+import android.app.Activity
+import androidx.activity.result.ActivityResultCaller
+import com.duckduckgo.app.statistics.pixels.Pixel
+import com.duckduckgo.di.scopes.ActivityScope
+import com.duckduckgo.voice.api.VoiceSearchLauncher.Event
+import com.duckduckgo.voice.api.VoiceSearchLauncher.Source
+import com.duckduckgo.voice.impl.ActivityResultLauncherWrapper.Action.LaunchVoiceSearch
+import com.duckduckgo.voice.impl.ActivityResultLauncherWrapper.Request
+import com.duckduckgo.voice.impl.listeningmode.ui.VoiceSearchBackgroundBlurRenderer
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+
+interface VoiceSearchActivityLauncher {
+    fun registerResultsCallback(
+        caller: ActivityResultCaller,
+        activity: Activity,
+        source: Source,
+        onEvent: (Event) -> Unit
+    )
+
+    fun launch()
+}
+
+@ContributesBinding(ActivityScope::class)
+class RealVoiceSearchActivityLauncher @Inject constructor(
+    private val blurRenderer: VoiceSearchBackgroundBlurRenderer,
+    private val pixel: Pixel,
+    private val activityResultLauncherWrapper: ActivityResultLauncherWrapper
+) : VoiceSearchActivityLauncher {
+
+    companion object {
+        private const val KEY_PARAM_SOURCE = "source"
+    }
+
+    private lateinit var _source: Source
+    private var _activity: Activity? = null
```

Comment:
This one's dealing with the blur in the callback, so maybe fire an event to remove the blur? Then this variable can also be removed.

Reply:
I wanted to put the blurring logic within the VoiceSearchLauncher logic so the features calling this API wouldn't have to do the blurring themselves and have repeated logic. I can still remove this variable by passing activity in the function params of launch.